[CI] MixedClusterEsqlSpecIT test {stats.StatsByMvFieldWhere SYNC} failing

**Build Scans:**
- [elasticsearch-periodic #11754 / 8.14.3_bwc](https://gradle-enterprise.elastic.co/s/7dhyzenkynaae)
- [elasticsearch-periodic #11754 / 8.14.3_bwc](https://gradle-enterprise.elastic.co/s/k3xus3phadrly)
- [elasticsearch-periodic #11744 / 8.14.3_bwc](https://gradle-enterprise.elastic.co/s/hh6ylwu5cmsu2)
- [elasticsearch-periodic #11729 / 8.14.3_bwc](https://gradle-enterprise.elastic.co/s/dozlwjnzwb56a)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:qa:server:mixed-cluster:v8.14.3#javaRestTest" -Dtests.class="org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT" -Dtests.method="test {stats.StatsByMvFieldWhere SYNC}" -Dtests.seed=E874ACD66759D40D -Dtests.bwc=true -Dtests.locale=lkt -Dtests.timezone=America/Merida -Druntime.java=25
```

**Applicable branches:**
8.19

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7Bstats.StatsByMvFieldWhere%20SYNC%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: Expected an empty list
0: <unexpected> but was "Line 3:9: evaluation of [salary_change > 1] failed, treating result as null. Only first 20 failures recorded."
1: <unexpected> but was "Line 3:9: java.lang.IllegalArgumentException: single-value function encountered multi-value"
```

**Issue Reasons:**
- [8.19] 2 consecutive failures in step 8.14.3_bwc
- [8.19] 4 failures in test test {stats.StatsByMvFieldWhere SYNC} (14.8% fail rate in 27 executions)
- [8.19] 4 failures in step 8.14.3_bwc (57.1% fail rate in 7 executions)
- [8.19] 3 failures in pipeline elasticsearch-periodic (50.0% fail rate in 6 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -331,15 +331,3 @@ tests:
 - class: org.elasticsearch.search.aggregations.metrics.TopHitsIT
   method: testMixedSortFieldTypes
   issue: https://github.com/elastic/elasticsearch/issues/142077
-- class: org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT
-  method: test {stats.StatsByMvFieldAggExpr}
-  issue: https://github.com/elastic/elasticsearch/issues/141835
-- class: org.elasticsearch.xpack.esql.ccq.MultiClusterSpecIT
-  method: test {stats.StatsByMvFieldWhere}
-  issue: https://github.com/elastic/elasticsearch/issues/140914
-- class: org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT
-  method: test {stats.StatsByMvFieldWhere SYNC}
-  issue: https://github.com/elastic/elasticsearch/issues/140915
-- class: org.elasticsearch.xpack.esql.qa.mixed.MixedClusterEsqlSpecIT
-  method: test {stats.StatsByMvFieldWhere ASYNC}
-  issue: https://github.com/elastic/elasticsearch/issues/140916

--- a/x-pack/plugin/esql/qa/testFixtures/src/main/resources/stats.csv-spec
+++ b/x-pack/plugin/esql/qa/testFixtures/src/main/resources/stats.csv-spec
@@ -3410,6 +3410,7 @@ SUM(a):long | a:integer | b:integer
 
 
 statsByMvFieldAggExpr
+required_capability: resolve_groupings_before_resolving_references_to_groupings_in_aggregations
 FROM employees 
 | STATS SUM(salary_change) + salary_change BY salary_change 
 | SORT salary_change 
@@ -3445,6 +3446,7 @@ SUM(salary_change):double   | salary_change:double  | x:double
 
 
 statsByMvFieldWhere
+required_capability: fix_stats_mv_constant_fold
 FROM employees 
 | STATS SUM(salary_change) BY salary_change 
 | WHERE salary_change > 1 

